File: string_search/str_search.c
#include <stdio.h>
#include <string.h>

/**
 * count_non_overlapping_occurrences - Counts non-overlapping occurrences of a substring in a string.
 * @main_str: The main string to search within.
 * @sub_str: The substring to search for.
 * 
 * Return: The count of non-overlapping occurrences.
 */
int countSubstrings(const char *main_str, const char *sub_str)
{
    if (!main_str || !sub_str || strlen(sub_str) == 0) {
        return 0;
    }

    int count = 0;
    const char *pos = main_str;

    while ((pos = strstr(pos, sub_str)) != NULL)
    {
        count++;
        pos += strlen(sub_str);
    }
    return count;
}

int main() 
{
    printf("%d\n", countSubstrings("ababab", "aba")); // Output: 1
    printf("%d\n", countSubstrings("abcabcabc", "abc")); // Output: 3
    printf("%d\n", countSubstrings("aaaaa", "aa")); // Output: 2
    printf("%d\n", countSubstrings("hello world", " ")); // Output: 1
    printf("%d\n", countSubstrings("hello world", "x")); // Output: 0
    return 0;
}

File: string_search/strSearch.py
#!/usr/bin/env python3
"""The module for String Searching"""


def countSubstrings(main_str, sub_str):
    """
    Count the number of non-overlapping occurrences of a substring in a string.

    Args:
    main_str (str): The string to search within.
    sub_str (str): The substring to search for.

    Returns:
    int: The count of non-overlapping occurrences.
    """
    if not sub_str or not main_str:
        return 0

    count = 0
    i = 0

    while i < len(main_str):
        next_index = main_str.find(sub_str, i)
        if next_index == -1:
            break
        count += 1
        i = next_index + len(sub_str)

    return count


print(countSubstrings("ababab", "aba"))
print(countSubstrings("abcabcabc", "abc"))
print(countSubstrings("aaaaa", "aa"))
print(countSubstrings("hello world", " "))
print(countSubstrings("hello world", "x"))
